File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.py
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        target = len(nums) / 2
        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
            if freq[num] == target:
                return num
        
        return -1

# array, hash table
# time: O(n)
# space: O(n)

File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.go
func repeatedNTimes(nums []int) int {
    target := len(nums) / 2
    freq := map[int]int{}
    for _, num := range nums {
        freq[num]++
        if freq[num] == target {
            return num
        }
    }

    return -1
}

// array, hash map
// time: O(n)
// space: O(n)